### PR TITLE
Parsing of lazy and force (Fixes #1510)

### LINKED ISSUE #1510: Precedence of `lazy` and `force`
Right now `force x == true` is parsed as `force (x == true)` which seems unexpected and impractical.

### PR DESCRIPTION
**Pull request by JonathanStarup**
Fixes #1510

Both lazy and force bind very tightly now, such that force/lazy 2 + 5 is parsed (force/lazy 2) + 5. The only places in the tests that broke because of this is in relation to acsribe, where lazy ENil : LazyList[Int32] was expected to be parsed lazy (ENil : LazyList[Int32]).

Should Ascribe be tighter than lazy/force?

### CODE REVIEW

**Comment by magnus-madsen** [comment]
@mlutze @jaschdoc  Thoughts? How would you expect `lazy <exp> : <type>` to be parsed?

What does OCaml do?

**Comment by jaschdoc** [comment]
Intuitively, I'd expect it to be parsed as `(lazy <exp>): <type>`.

**Comment by mlutze** [comment]
Seems like OCaml [requires ascriptions to be parenthesized](https://ocaml.org/manual/expr.html#expr).

I agree with @jaschdoc that it _feels_ like `lazy` should bind more tightly than ascription.

**Comment by magnus-madsen** [comment]
It seems there is agreement and if I understood correctly, this PR already implements what is wanted. Correct?

File: main/test/flix/Test.Exp.Force.flix [hunk 1]
```diff
@@ -160,6 +160,12 @@ namespace Test/Exp/Force {
         let x = lazy None;
         None: Poly[Unit, Unit] == force x
 
+    @test
+    def testPrecedenceForce01(): Bool =
+        let x = lazy 2;
+        let y = force x + 1;
+        y == 3 and force x == 2
+
```
Comment (magnus-madsen) [inline main/test/flix/Test.Exp.Force.flix:168]: Could I ask for a few more tests, say 3-5-10. They can be very similar. Perhaps with operators of different precedence.

Comment (JonathanStarup) [reply main/test/flix/Test.Exp.Force.flix:168]: What do you think about merging the lazy/force test files? the force test file sort of have to test lazy anyway

**Comment by JonathanStarup** [comment]
> It seems there is agreement and if I understood correctly, this PR already implements what is wanted. Correct?

Yes, and now there's a couple more tests

**Comment by magnus-madsen** [comment]
Thanks!

**Pull request merged**